### DISCUSSION TITLE
Hypre Boomerang Simulation freezes

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#22367] [Q&A General] [answered]
I've recently encountered an issue while using Hypre Boomerang as my petsc options, similar to Q+A #17058 but not exactly the same.

In that thread the user found that removing the Chebyshev relaxation they resolved their issue, however my HYPRE options were just:
```
line_search = 'none'
petsc_options = '-snes_ksp_ew'
petsc_options_iname = '-ksp_gmres_restart -pc_type -pc_hypre_type -pc_hypre_boomeramg_max_iter'
petsc_options_value = '2001                hypre    boomeramg      20'
```
Which was causing hang-ups/freezes during nonlinear/linear iterations (it would print a nonlinear iteration then just stop, so maybe on the transition from NL to L?) but often not until an hour or more into the simulation. Running top on the nodes involved in the job would reveal that while their CPU usage was still maxed out, they were no longer using any memory at all. I don't believe memory was the issue, as I was only running around 2000 local NL DOFs so there was not much information being stored on each node.

I'm not very familiar with debugging at all, and fear that given it takes about an hour to fail on opt mode it would take a very long time to reach failure in debug. However, I believe the issue was hypre boomeramg as a run on petsc ilu with the following options:
```
line_search = 'default'
petsc_options_iname = '-pc_type  -snes_linesearch_type -pc_factor_shift_type -snes_max_it -ksp_max_it'
petsc_options_value = 'ilu        basic                 NONZERO               20           40'
```
Has just lasted several hours without issue.
Interestingly, I also find that disabling distributed mesh while still using HYPRE BOOMERANG also fixes the freezing issue, so I'm not sure what the correlation there is.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Do you not get any live print?
Can you turn on the linear residual printing? The max linear iterations is 10k so that can feel like a freeze
In Outputs: print_linear_residual = true


Guillaume

**Reply by Edward-Eth** [reply]
Linear residual printing is on, the simulation normally takes about 20 seconds for a timestep but will freeze indefinitely (I've tried waiting multiple hours) after printing a nonlinear iteration.

**Reply by GiudGiud** [reply]
can you please paste the input file here?

**Reply by Edward-Eth** [reply]
I've been attempting to make an MWE, will post if I manage to make one that's simple/doesn't contain any information.

**Reply by GiudGiud** [reply]
Any update on this?

**Reply by Edward-Eth** [reply]
So I've not been able to make an MWE that I'm able to share here, but I managed to "resolve" the issue by totally removing the petsc_options, petsc_options_iname, petsc_options_value lines. I'm not sure what petsc options this means I'm now running? I was also observing freezes when using HMG as my -pc_type

**Reply by lindsayad** [reply]
ILU in serial, block Jacobi with ILU on the sub-blocks in parallel.

You can always run with the PETSc option `-snes_view` which will show all the information about your nonlinear solution algorithm in PETSC including the linear solution algorithm, or you can pass `-ksp_view` to just get info about the linear solve (although you'll get a printout for every linear solve)
